File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<pair<int,int>> v(nums.size());
        for(int i=0;i<nums.size();i++){
            v[i].first=nums[i];
            v[i].second=i;
        }
        int n=nums.size(),i=0,j=n-1;
        sort(v.begin(),v.end());
       
        // vector<int> ans;
        while(i<j){
            if(v[i].first+v[j].first==target){
                return {v[i].second,v[j].second};
            }
            else if(v[i].first+v[j].first>target)j--;
            else i++;
        }
        return {};
    }
};

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var x=[];
    for(var i=0;i<nums.length;i++){
        x.push([nums[i],i]);
    }
    x.sort((a,b)=>a[0]-b[0]);
    let l=0,r=nums.length-1;
    while(l<r){
        var sum=x[l][0]+x[r][0]
        if(sum==target){
            return [x[l][1],x[r][1]];
        }
        else if(sum<target){
            l++;
        }
        else{
            r--;
        }
    }
    return [];
};